omod/src/main/java/org/openmrs/web/controller/user/UserFormController.java

```diff
@@ -112,6 +115,15 @@ public List<Role> getRoles(WebRequest request) {
 	public String showForm(@RequestParam(required = false, value = "userId") Integer userId,
 	        @RequestParam(required = false, value = "createNewPerson") String createNewPerson,
 	        @ModelAttribute("user") User user, ModelMap model) {
+
+		try {
+			Field emailField = getEmailField();
```

Comment:
Why don't you get the email field only for the platform versions that support it?

Reply:
i was trying to avoid duplicating the code ` User.class.getDeclaredField("email")`  when; 1) determining if the platform version actually supports this field, and, 2) when i actually need access the email filed to set it

I have another way though.  It might not be ideal though  using `OPENMRS_VERSION_SHORT`
```
	private boolean isPlatformTwoPointTwoOrAbove() {
        String platformVersion = OpenmrsConstants.OPENMRS_VERSION_SHORT.substring(0, 3);

        try {
            float versionFloat = Float.parseFloat(platformVersion);
			if (versionFloat>=2.2) {
				return true;
			}
        } catch (NumberFormatException e) {
			log.error("Unable to parse platform value text to float", e);

        }
		return false;
    }
    ```